### PR TITLE
Add option for a `history` button.

### PR DESCRIPTION
**Pull request by noamross**
This PR adds the ability to add a "History" button, similar to an "Edit" button, as part of custom configuration.  It essentially duplicates the code associated with the edit button code so that you can get something like this:

screenshot 2018-09-29 16 38 14

I've included changes to the gitbook documentation but not in the examples.

This addresses #638 from @maelle. I note I previously signed an individual contributor agreement on 2017-07-28.

### CODE REVIEW

**Comment by noamross** [comment]
I've collapsed those functions to `source_link()` and `source_link_setting()`, both with a 'type' option, squashed to a single commit and rebased to master.

File: R/gitbook.R [hunk 1]
```diff
@@ -138,10 +138,16 @@ gitbook_page = function(
   }
 
   # you can set the edit setting in either _bookdown.yml or _output.yml
-  if (is.list(setting <- edit_setting(config$edit))) config$edit = setting
+  if (is.list(setting <- source_link_setting(config$edit, type="edit")))
```
Comment (yihui) [inline R/gitbook.R]: nit: spaces around `=`

File: R/gitbook.R [hunk 2]
```diff
@@ -138,10 +138,16 @@ gitbook_page = function(
...
+    config$edit = setting
   if (length(rmd_cur) && is.list(config$edit))
     config$edit$link = sprintf(config$edit$link, rmd_cur)
 
+  if (is.list(setting <- source_link_setting(config$history, type="history")))
+    config$history = setting
+  if (length(rmd_cur) && is.list(config$history))
+    config$history$link = sprintf(config$history$link, rmd_cur)
```
Comment (yihui) [inline R/gitbook.R]: I think the above two blocks of code can also be collapsed into a for loop. They look too similar.

```r
for (type in c('edit', 'history')) {
  ...
}
```

File: R/html.R [hunk 3]
```diff
@@ -430,21 +431,21 @@ button_link = function(target, text) {
   )
 }
 
-edit_link = function(target) {
+source_link <- function(target, type) {
```
Comment (yihui) [inline R/html.R]: `<-` -> `=`

**Comment by noamross** [comment]
Sorry for the delay! Just addressed these changes and collapsed to a single commit.

**Pull request merged**